src/server/svr_func.c

```diff
@@ -4931,15 +4931,27 @@ find_prov_vnode_list(job *pjob, exec_vnode_listtype *prov_vnodes, char **aoe_nam
 					DBPRT(("%s: %s\n", __func__, (*prov_vnodes)[i]))
 					++i;
 					if (aoe_name != NULL) {
-						aoe = malloc(strlen(((pkvp + k)->kv_val)) + 1);
-						if (aoe == NULL) {
-							free(sbuf);
-							free(pbuf);
-							return -1;
+						if (*aoe_name) {
+							if (strcmp(*aoe_name, ((pkvp + k)->kv_val)) != 0) {
+								/* Aoe name can not be different across chunks, it's an error */
+								free(sbuf);
+								free(pbuf);
+								free(*aoe_name);
+								return -1;
+							}
+						} else {
+							aoe = malloc(strlen(((pkvp + k)->kv_val)) + 1);
+							if (aoe == NULL) {
+								free(sbuf);
+								free(pbuf);
+								if (*aoe_name)
+									free(*aoe_name);
```

Comment:
*aoe_name will be NULL.  This is the else case of "if (*aoe_name)"

Reply:
Right, this code is unnecessary